src/rpc/daemon_handler.cpp

```diff
@@ -0,0 +1,726 @@
+    {
+      res.status = Message::STATUS_FAILED;
+      res.error_details = "core::get_tx_outputs_gindexs() returned false";
+      return;
+    }
+
+    res.status = Message::STATUS_OK;
+
+  }
+
+  //TODO: handle "restricted" RPC
+  void DaemonHandler::handle(GetRandomOutputsForAmounts::Request& req, GetRandomOutputsForAmounts::Response& res)
+  {
+    auto& chain = m_core.get_blockchain_storage();
+
+    for (uint64_t& amount : req.amounts)
+    {
+      std::vector<uint64_t> indices = chain.get_random_outputs(amount, req.count);
+
+      outputs_for_amount ofa;
+
+      ofa.resize(indices.size());
+
+      for (uint64_t i = 0; i < indices.size(); i++)
+      {
+        crypto::public_key key = chain.get_output_key(amount, indices[i]);
+        ofa[i].amount_index = indices[i];
+        ofa[i].key = key;
+      }
+
+      amount_with_random_outputs amt;
+      amt.amount = amount;
+      amt.outputs = ofa;
+
+      res.amounts_with_outputs.push_back(amt);
+    }
+
+    res.status = Message::STATUS_OK;
+  }
+
+  void DaemonHandler::handle(SendRawTx::Request& req, SendRawTx::Response& res)
+  {
+    auto tx_blob = cryptonote::tx_to_blob(req.tx);
+
+    cryptonote_connection_context fake_context = AUTO_VAL_INIT(fake_context);
+    tx_verification_context tvc = AUTO_VAL_INIT(tvc);
+
+    if(!m_core.handle_incoming_tx(tx_blob, tvc, false, false) || tvc.m_verifivation_failed)
+    {
+      if (tvc.m_verifivation_failed)
+      {
+        LOG_PRINT_L0("[on_send_raw_tx]: tx verification failed");
+      }
+      else
+      {
+        LOG_PRINT_L0("[on_send_raw_tx]: Failed to process tx");
+      }
+      res.status = Message::STATUS_FAILED;
+
+      //TODO: these should be mutually exclusive, as encountering one
+      //      cancels the verification process.  Need to confirm this.
+      if (tvc.m_low_mixin)
+        res.error_details = "mixin too low";
+      else if (tvc.m_double_spend)
+        res.error_details = "double spend";
+      else if (tvc.m_invalid_input)
+        res.error_details = "invalid input";
+      else if (tvc.m_invalid_output)
+        res.error_details = "invalid output";
+      else if (tvc.m_too_big)
+        res.error_details = "too big";
+      else if (tvc.m_overspend)
+        res.error_details = "overspend";
+      else if (tvc.m_fee_too_low)
+        res.error_details = "fee too low";
+      else if (tvc.m_not_rct)
+        res.error_details = "tx is not ringct";
+      else
+        res.error_details = "an unknown issue was found with the transaction";
+
+      return;
+    }
+
+    if(!tvc.m_should_be_relayed || !req.relay)
+    {
+      LOG_PRINT_L0("[on_send_raw_tx]: tx accepted, but not relayed");
+      res.error_details = "Not relayed";
+      res.relayed = false;
+      res.status = Message::STATUS_OK;
+
+      return;
+    }
+
+    NOTIFY_NEW_TRANSACTIONS::request r;
+    r.txs.push_back(tx_blob);
+    m_core.get_protocol()->relay_transactions(r, fake_context);
```

Comment:
followed the rabbit hole a few levels deep -- always returns true.  There's a lot of that...